platform/linux-generic/arch/aarch64/cpu_flags.c

```diff
@@ -8,11 +8,738 @@
 
 #include <string.h>
 #include <stdlib.h>
+#include <sys/auxv.h>
+#include <asm/hwcap.h>
 
 #include <odp/api/hints.h>
 #include <odp_debug_internal.h>
 #include "cpu_flags.h"
 
+typedef struct {
+	const char *feat_flag;
+	const unsigned int hwcap_field;
+	bool valid;
+} hwcap_feat_flag_t;
+
+static hwcap_feat_flag_t hwcap_flags[] = {
+#ifdef HWCAP_FP
+	{
+		/* Floating-point Extensions */
+		.feat_flag = "FEAT_FP",
+		.hwcap_field = HWCAP_FP,
+		.valid = 1,
+	},
+#else
+	{
+		.feat_flag = "FEAT_FP",
+		.valid = 0,
+	},
+#endif
+
+#ifdef HWCAP_ASIMD
```

Comment:
As with compile time flags, it would be easier to check the output if the feature flags would be in alphabetical order.

Reply:
I had written the implementation to be consistent with the Linux hwcap.h file: https://github.com/torvalds/linux/blob/master/arch/arm64/include/uapi/asm/hwcap.h. It makes a lot more sense this way because the array indexing matches exactly with macro definitions, i.e, 5th flag, `HWCAP_PMULL` is at location `hwcap_flags[4] `. 
Let me know what you think. Thanks!

Reply:
Also, I just saw that the output of `lscpu` is also in the same order as given in the `hwcap.h` file

Reply:
Okay, that makes sense.